src/cl_ents.c

```diff
@@ -2062,20 +2063,23 @@ static void CL_LinkPlayers(void)
 			}
 		}
 
-		VectorCopy(cent->lerp_origin, end);
-		end[2] += 12;
-		trace = PM_TraceLine(r_refdef.vieworg, end);
+		if (gl_spec_xray.value && (cls.demoplayback || cls.mvdplayback))
```

Comment:
This only fixes the issue with gameplay. While it's the most acute, doing something like what `PF_tracebox` does would probably be the real fix, unless I'm misinterpreting what it does.. what I mean is that instead tracing the 8 corners of the entity, and as it's only used outside of actual gameplay, the added cost is irrelevant, and it's probably very cheap. Doing so also makes the `end[2]` tweak redundant.